### PR TITLE
(#91) The lockscreen-app no longer responds to on_lock messages if it is has already locked

### LINKED ISSUE #91: Stop respawning `lockscreen-app` un unlock
If the lockscreen app happens to have a problem and crashes on startup (but after it's able to acquire the screen lock), Miriway will continue respawning it even after the session is unlocked through logind.

This means the session becomes unusable, as unlocking it just triggers another lock with the respawned app.

### PR DESCRIPTION
**Pull request by mattkae**
fixes #91

### CODE REVIEW

**Review by AlanGriffiths — COMMENTED** [review]
The description seems wrong as there is no change to the lockscreen-app.

I think what this does is "if the lockscreen crashes then only relaunch it when the session is **unlocked**".

But, is this solving the right problem? Wouldn't the right behaviour be "if the lockscreen crashes then only relaunch it when the session is **locked**"?

**Comment by mattkae** [comment]
> The description seems wrong as there is no change to the lockscreen-app.
> 
> I think what this does is "if the lockscreen crashes then only relaunch it when the session is **unlocked**".

If the lockscreen crashes, then the lockscreen will _still_ relaunch, since that behavior is confined to `shell_launch`. The `is_locked` check only occurs on the initial launch.

> But, is this solving the right problem? Wouldn't the right behaviour be "if the lockscreen crashes then only relaunch it when the session is **locked**"?

This fixes the following problem:
1. The session is locked
2. The lockscreen client starts up
3. The lockscreen client acquires the lock, BUT it dies right afterward.
4. In that time, our internal `on_lock` notification got sent out, which requests for the lockscreen client to start
5. However, the _previous_ lockscreen client is already in the process of restarting, so we should _not_ try to start another one

File: miriway.cpp [hunk 1]
```diff
@@ -131,13 +131,23 @@ class LockScreen
         lockscreen_option(server);
         session_locker(server);
     }
+
+    bool is_locked() const { return is_locked_; }
+
 private:
     std::function<void(std::vector<std::string> const& command_line)> const launch;
     ConfigurationOption const lockscreen_option;
+    bool is_locked_ = false;
```
Comment (AlanGriffiths) [inline miriway.cpp:140]: This needs to be `atomic<>`, it is accessed from multiple threads

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]

**Comment by mattkae** [comment]
@AlanGriffiths After speaking with saviq, I officially take back this PR :) What we want to do is only retry creating the lockscreen if we're locked. Over over thought this one. I will close and reopen

**Pull request closed without merging**